src/models/principal.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * CDAV Library
+ *
+ * This library is part of the Nextcloud project
+ *
+ * @author Georg Ehrke
+ * @copyright 2018 Georg Ehrke <oc.list@georgehrke.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public
+ * License along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import { DavObject } from './davObject.js';
+import * as NS from '../utility/namespaceUtility.js';
+
+/**
+ * @class
+ */
+export class Principal extends DavObject {
+
+	/**
+	 * Creates an object that represents a single principal
+	 * as specified in RFC 3744
+	 *
+	 * https://tools.ietf.org/html/rfc3744#section-2
+	 *
+	 * @inheritDoc
+	 */
+	constructor(...args) {
+		super(...args);
+
+		super._exposeProperty('displayname', NS.DAV, 'displayname');
```

Comment:
We use pascalCase everywhere but here, should we?
`displayName`

Reply:
It's `displayname` in `DavCollection.js` as well, so i wanted to keep it consistent:
https://github.com/nextcloud/cdav-library/blob/master/src/models/davCollection.js#L70